### PR TITLE
Decouple the deinitialization detection from `Lifetime`.

### PR DESCRIPTION
**Pull request by andersio**
Originally #3112.
Solves #3111.

-

> Instead of having magic strings I'd recommend storing the events themselves. Otherwise this test is pretty fragile.

Since `Event` is not equatable in this case, this is probably why @sharplet had written the test in such way. Anyway, I have made them to now check only for completion, which is what the lifetime contract is all about.

### CODE REVIEW

File: ReactiveCocoaTests/Swift/LifetimeSpec.swift [hunk 1]
```diff
@@ -42,27 +42,25 @@ final class LifetimeSpec: QuickSpec {
...
-					case .failed:
-						events.append("failed")
-					case .interrupted:
-						events.append("interrupted")
-					}
-				}
+				var isCompleted = false
+				lifetime.ended.observeCompleted { isCompleted = true }
 
 				object.value = nil
```
Comment (NachoSoto) [inline ReactiveCocoaTests/Swift/LifetimeSpec.swift]: I would check that  `isCompleted` is `false` here.

**Comment by NachoSoto** [comment]
:shipit:

File: ReactiveCocoaTests/Swift/LifetimeSpec.swift (hunk 1 above)
Comment (andersio) [inline ReactiveCocoaTests/Swift/LifetimeSpec.swift]: observation should have no side effect, shouldn't it?

File: ReactiveCocoa/Swift/Lifetime.swift [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
-		(ended, endedObserver) = Signal.pipe()
+	/// Initialize a `Lifetime` from a lifetime token, which is expected to be
+	/// associated with an object.
+	///
+	/// -important: The resulting lifetime object does not retain the lifetime
+	///             token.
+	///
+	/// -parameters:
+	///  -token: A lifetime token for detecting the deinitialization of the
+	///          associated object.
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Lifetime.swift]: There should be a ` ` after each `-`s here.

File: ReactiveCocoaTests/Swift/LifetimeSpec.swift (hunk 1 above)
Comment (mdiep) [inline ReactiveCocoaTests/Swift/LifetimeSpec.swift]: I think it mostly makes it a bit clearer that you're testing that `object.value = nil` is expected to trigger `isCompleted = true`. But I can envision a bug where the completed block could complete immediately.

**Comment by mdiep** [comment]
Needs a merge, then it's ready to 🚢.

**Pull request merged**